python/cuml/cuml/internals/base.pyx

```diff
@@ -851,3 +851,63 @@ class UniversalBase(Base):
                 raise ex
 
             raise ex
+
+    def as_sklearn(self):
+        """
+        Convert the current GPU-accelerated estimator into a scikit-learn estimator.
+        This method imports and builds an equivalent CPU-backed scikit-learn model,
+        transferring all necessary parameters from the GPU representation to the
+        CPU model. After this conversion, the returned object should be a fully
+        compatible scikit-learn estimator, allowing you to use it in standard
+        scikit-learn pipelines and workflows.
+
+        Returns
+        -------
+        sklearn.base.BaseEstimator
+            A scikit-learn compatible estimator instance that mirrors the trained
+            state of the current GPU-accelerated estimator.
+
+        """
+        self.import_cpu_model()
+        self.build_cpu_model()
+        self.gpu_to_cpu()
+        return self._cpu_model
+
+    @classmethod
+    def from_sklearn(cls, model):
```

Comment:
Would be great to have a global conversion table, so that we don't need to provide its class as a parameter.

Reply:
This is a class method, so we get the class from that, it's not something the user passes (like `self` in non class methods) 

A global conversion table will be useful for a follow up to add `cuml.from_sklearn` library type of functionality though